server/tso/allocator_manager.go

```diff
@@ -561,9 +581,25 @@ func (am *AllocatorManager) ClusterDCLocationChecker() {
 				log.Warn("get or create the local tso suffix failed", zap.String("dc-location", dcLocation), errs.ZapError(err))
 				continue
 			}
+			if suffix > am.mu.maxSuffix {
+				// Because the number of suffix bits is changing dynamically according to the dc-location number,
+				// there is a corner case may cause the Local TSO is not unique while member changing.
+				// Example:
+				//     t1: xxxxxxxxxxxxxxx1 | 11
+				//     t2: xxxxxxxxxxxxxxx | 111
+				// Please take a look at https://github.com/tikv/pd/issues/3260 for more details.
+				if CalSuffixBits(suffix) != CalSuffixBits(am.mu.maxSuffix) {
+					suffixBitsChanged = true
+				}
+				am.mu.maxSuffix = suffix
+			}
 			am.mu.clusterDCLocations[dcLocation].suffix = suffix
 		}
 	}
+	am.mu.Unlock()
+	if suffixBitsChanged {
+		am.syncGlobalTSO()
```

Comment:
Is it possible to generate local tso before `syncGlobalTSO`? the suffix already generated before `syncGlobalTSO`.

Reply:
Theoretically yes, but it's very unlikely to occur this situation. Because PD leader will sync with the local as soon as the Local TSO Allocator leader is elected out. This should happen in a second. Consdering the PD client won't spot the new Local TSO Allocator so qucikly. I thinks it acceptable (PD client will update the PD leader and Local TSO Allocators every minute).

Reply:
What if there is something wrong when syncGlobalTSO? How long should we wait before the next check?

Reply:
I think we should make `syncGlobalTSO` succeed before providing the TSO service in the newly dc allocator. BTW, Does the member delete not adjust the max suffix?

Reply:
@nolouch For now, we don't recall the suffix has already allocated, so the member deleting will not affect the max suffix.